web/src/hooks/useDAGChart.ts

```diff
@@ -1,30 +1,28 @@
 import * as d3DAG from 'd3-dag';
-import {Dag} from 'd3-dag';
-import _ from 'lodash';
-import {TSpanMap} from '../components/Diagram/components/DAG';
+import {useMemo} from 'react';
 
-interface DAGResponse {
-  dag: Dag<{id: string; parentIds: string[]}, undefined>;
-  layout: {width: number; height: number};
+export interface IDAGNode<T> {
+  data: T;
+  id: string;
+  parentIds: string[];
 }
 
-export const useDAGChart = (spanMap: TSpanMap = {}): void | DAGResponse => {
```

Comment:
Instead of having this tightly coupled to the span map now it receives a list of nodes so the hook can be used from multiple components.